src/main.cpp

```diff
@@ -12,121 +12,83 @@ All rights reserved (see LICENSE).
 
 #if USE_LIBOSRM
 #include "osrm/exception.hpp"
 #endif
 
+#include "../include/cxxopts/include/cxxopts.hpp"
 #include "problems/vrp.h"
 #include "structures/cl_args.h"
 #include "utils/helpers.h"
 #include "utils/input_parser.h"
 #include "utils/output_json.h"
 #include "utils/version.h"
 
-void display_usage() {
-  std::string usage = "VROOM Copyright (C) 2015-2021, Julien Coupey\n";
-  usage += "Version: " + vroom::get_version() + "\n";
-  usage += "Usage:\n\tvroom [OPTION]... \"INPUT\"";
-  usage += "\n\tvroom [OPTION]... -i FILE\n";
-  usage += "\tvroom [OPTION]...\n";
-  usage += "Options:\n";
-  usage += "\t-a PROFILE:HOST (=" + vroom::DEFAULT_PROFILE +
-           ":0.0.0.0)\t routing server\n";
-  usage += "\t-c,\t\t\t\t choose ETA for custom routes and report violations\n";
-  usage += "\t-g,\t\t\t\t add detailed route geometry and indicators\n";
-  usage += "\t-i FILE,\t\t\t read input from FILE rather than from stdin\n";
-  usage += "\t-l LIMIT,\t\t\t stop solving process after LIMIT seconds\n";
-  usage += "\t-o OUTPUT,\t\t\t output file name\n";
-  usage += "\t-p PROFILE:PORT (=" + vroom::DEFAULT_PROFILE +
-           ":5000),\t routing server port\n";
-  usage += "\t-r ROUTER (=osrm),\t\t osrm, libosrm, ors or valhalla\n";
-  usage += "\t-t THREADS (=4),\t\t number of threads to use\n";
-  usage += "\t-x EXPLORE (=5),\t\t exploration level to use (0..5)";
-  std::cout << usage << std::endl;
-  exit(0);
-}
-
 int main(int argc, char** argv) {
-  // Load default command-line options.
   vroom::io::CLArgs cl_args;
-
-  // Parsing command-line arguments.
-  const char* optString = "a:ce:gi:l:o:p:r:t:x:h?";
-  int opt = getopt(argc, argv, optString);
-
+  std::string host_arg;
+  std::string port_arg;
   std::string router_arg;
-  std::string limit_arg = "";
-  std::string nb_threads_arg = std::to_string(cl_args.nb_threads);
-  std::string exploration_level_arg = std::to_string(cl_args.exploration_level);
+  std::string limit_arg;
+  std::string nb_threads_arg;
   std::vector<std::string> heuristic_params_arg;
 
-  while (opt != -1) {
-    switch (opt) {
-    case 'a':
-      vroom::io::update_host(cl_args.servers, optarg);
-      break;
-    case 'c':
-      cl_args.check = true;
-      break;
-    case 'e':
-      heuristic_params_arg.push_back(optarg);
-      break;
-    case 'g':
-      cl_args.geometry = true;
-      break;
-    case 'h':
-      display_usage();
-      break;
-    case 'i':
-      cl_args.input_file = optarg;
-      break;
-    case 'l':
-      limit_arg = optarg;
-      break;
-    case 'o':
-      cl_args.output_file = optarg;
-      break;
-    case 'p':
-      vroom::io::update_port(cl_args.servers, optarg);
-      break;
-    case 'r':
-      router_arg = optarg;
-      break;
-    case 't':
-      nb_threads_arg = optarg;
-      break;
-    case 'x':
-      exploration_level_arg = optarg;
-      break;
-    default:
-      break;
-    }
-    opt = getopt(argc, argv, optString);
+  // clang-format off
```

Comment:
I agree clang-format is especially ugly on the `options....` part, but what about:

- still clang-formatting the `options` definition and the `add_options("debug_group")` part;
- avoid overly long lines by using one argument per line for some of the options.

Reply:
hm. from my experience that's not how clang-format works, i.e. you can't just set up your own formatting here (one arg per line), would also kinda defeat the whole purpose of it I guess. I couldn't verify since I'm on arch linux and have only v13, not v10 available. the last commit has the changes you proposed, could you run and try for yourself? 

side note, in valhalla we have a kinda decent way to detach the clang-format version from individual package managers: https://github.com/jarro2783/cxxopts/tags